File: python/1695.maximumUniqueSubarray.py
#!/usr/bin/env python3
class Solution:
    def maximumUniqueSubarray(self, nums: list[int]) -> int:
        numSet = set()
        left,highest,s = 0,0,0
        for right in range(len(nums)):
            if nums[right] not in numSet:
                numSet.add(nums[right])
                s += nums[right]
            else:
                while nums[left] != nums[right]:
                    numSet.remove(nums[left])
                    s -= s[left]
                    left += 1
                left += 1
            if s > highest:
                highest = s
        return highest

test = Solution()
print(test.maximumUniqueSubarray([1,1,1,1]))

File: java/1695.maximumUniqueSubarray.java
package java;

import java.util.HashSet;
import java.util.Set;

public class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        Set<Integer> numSet = new HashSet<Integer>();
        int left = 0;
        int sum = 0;
        int highest = 0;
        for(int right = 0;right < nums.length; right++){
            if(!numSet.contains(nums[right])){
                numSet.add(nums[right]);
                sum += nums[right];
            }
            else{
                while(nums[left] != nums[right]){
                    numSet.remove(nums[left]);
                    sum -= nums[left];
                    left++;
                }
                left++;
            }
            if(sum > highest){
                highest = sum;
            }
        }
        return highest;
    }
}
